integration-tests/relayinterface/lookups_test.go

```diff
@@ -117,12 +125,201 @@ func TestAccountLookups(t *testing.T) {
+			IsSigner:   chainwriter.MetaBool{Value: true},
+			IsWritable: chainwriter.MetaBool{Value: true},
 		}
 		_, err := lookupConfig.Resolve(ctx, testArgs, nil, nil)
 		require.Error(t, err)
 	})
+
+	t.Run("AccountLookup works with MetaBool lookups", func(t *testing.T) {
+		accounts := [3]*solana.AccountMeta{}
+
+		for i := 0; i < 3; i++ {
+			accounts[i] = &solana.AccountMeta{
+				PublicKey:  chainwriter.GetRandomPubKey(t),
+				IsSigner:   true,
+				IsWritable: false,
+			}
+		}
+
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "InvalidAccount",
+			Location:   "Inner.Accounts.PublicKey",
+			IsSigner:   chainwriter.MetaBool{Location: "Inner.Accounts.IsSigner"},
+			IsWritable: chainwriter.MetaBool{Location: "Inner.Accounts.IsWritable"},
+		}
+
+		args := TestAccountArgs{
+			Inner: InnerAccountArgs{
+				Accounts: accounts[:],
+			},
+		}
+
+		result, err := lookupConfig.Resolve(ctx, args, nil, nil)
+		require.NoError(t, err)
+
+		for i, meta := range result {
+			require.Equal(t, accounts[i], meta)
+		}
+	})
+
+	t.Run("AccountLookup works with MetaBool lookups when a meta field is missing", func(t *testing.T) {
+		accounts := [3]*solana.AccountMeta{}
+
+		for i := 0; i < 3; i++ {
+			accounts[i] = &solana.AccountMeta{
+				PublicKey:  chainwriter.GetRandomPubKey(t),
+				IsWritable: true,
+			}
+		}
+
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "InvalidAccount",
+			Location:   "Inner.Accounts.PublicKey",
+			IsSigner:   chainwriter.MetaBool{Location: "Inner.Accounts.IsSigner"},
+			IsWritable: chainwriter.MetaBool{Location: "Inner.Accounts.IsWritable"},
+		}
+
+		args := TestAccountArgs{
+			Inner: InnerAccountArgs{
+				Accounts: accounts[:],
+			},
+		}
+
+		result, err := lookupConfig.Resolve(ctx, args, nil, nil)
+		require.NoError(t, err)
+
+		for i, meta := range result {
+			require.Equal(t, accounts[i], meta)
+		}
+	})
+
+	t.Run("AccountLookup works with MetaBool lookups in a different location", func(t *testing.T) {
+		type TestAccountArgsExtended struct {
+			Inner        InnerAccountArgs
+			ExternalBool bool
+		}
+
+		accounts := [3]*solana.AccountMeta{}
+
+		for i := 0; i < 3; i++ {
+			accounts[i] = &solana.AccountMeta{
+				PublicKey:  chainwriter.GetRandomPubKey(t),
+				IsWritable: true,
+				IsSigner:   true,
+			}
+		}
+
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "InvalidAccount",
+			Location:   "Inner.Accounts.PublicKey",
+			IsSigner:   chainwriter.MetaBool{Location: "ExternalBool"},
+			IsWritable: chainwriter.MetaBool{Location: "ExternalBool"},
+		}
+
+		args := TestAccountArgsExtended{
+			Inner: InnerAccountArgs{
+				Accounts: accounts[:],
+			},
+			ExternalBool: true,
+		}
+
+		result, err := lookupConfig.Resolve(ctx, args, nil, nil)
+		require.NoError(t, err)
+
+		for i, meta := range result {
+			require.Equal(t, accounts[i], meta)
+		}
+	})
+
+	t.Run("AccountLookup fails with MetaBool an invalid number of Meta lookups", func(t *testing.T) {
```

Comment:
nit: think the description is missing something? Also could you add the success scenario for this case too just so we can make sure the correct index bool is applied to the correct account? 
```suggestion
	t.Run("AccountLookup fails with MetaBool due to an invalid number of Meta lookups", func(t *testing.T) {
```

Reply:
Fixed the description, but isn't that basically what some of the above tests are checking already? I added [this](https://github.com/smartcontractkit/chainlink-solana/blob/47f38136b823777243a6d5d849f2c33ed259561e/integration-tests/relayinterface/lookups_test.go#L141-L142) to the first test to ensure the order is being checked

Reply:
Oh right, I was thinking the first test was different since it was looking back into the parent struct and this was an external field.